// The Firmament project
// Copyright (c) 2011-2012 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
//
// TaskDelegationMessage is used by a coordinator to delegate rsource

package firmament;

import "base/task_desc.proto";

message TaskDelegationRequestMessage {
  required TaskDescriptor task_descriptor = 2;
  optional bytes target_resource_id = 1;
  optional bytes delegating_resource_id = 3;
}

message TaskDelegationResponseMessage {
  required uint64 task_id = 1;
  required bool success = 2;
  optional bytes target_resource_id = 3;
}
